parts/k8s/cloud-init/artifacts/cse_config.sh

```diff
@@ -622,10 +622,12 @@ configGPUDrivers() {
   echo blacklist nouveau >>/etc/modprobe.d/blacklist.conf
   retrycmd_no_stats 120 5 25 update-initramfs -u || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
   wait_for_apt_locks
-  dpkg -i $(ls ${APT_CACHE_DIR}libnvidia-container1*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
-  dpkg -i $(ls ${APT_CACHE_DIR}libnvidia-container-tools*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
-  dpkg -i $(ls ${APT_CACHE_DIR}nvidia-container-toolkit*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
-  dpkg -i $(ls ${APT_CACHE_DIR}nvidia-container-runtime*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
+  for apt_package in $NVIDIA_PACKAGES; do
+    dpkg -i $(ls ${PERMANENT_CACHE_DIR}${apt_package}*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
+    dpkg -i $(ls ${PERMANENT_CACHE_DIR}${apt_package}*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
+    dpkg -i $(ls ${PERMANENT_CACHE_DIR}${apt_package}*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
+  done
+  dpkg -i $(ls ${PERMANENT_CACHE_DIR}nvidia-container-runtime*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
```

Comment:
I note here that you don't do 3 tries of this (which is likely just fine anyway since the above 3 tries don't seem to make sense)

Again, don't understand the use of `$(ls ...)` here.

Reply:
That's because we are installing the `nvidia-container-runtime` package here, and not enumerating over an "array of packages".

The reason we do the "array of 3 packages" plus "one single package" is because the former array of packages all use a common release, and an enumeration allows us to enforce that version consistency.

The `ls` thing is actually a glob in order to produce the literal package filename that's on disk as input to `dpkg -i`.

Reply:
The ls and the glob produces the exact same results - seriously.  The only difference is that the ls does it as a string that is then parsed by the shell into argv and the glob does it in the shell and then passed as argv directly.

Reply:
You're saying that using `dpkg -i ${PERMANENT_CACHE_DIR}nvidia-container-runtime*` will similarly be processed via the glob interpreter to yield a filename (or list of filenames) as the input to dpkg?...

Reply:
Yes